activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -110,6 +110,30 @@ def index_by
     end
   end
 
+  # Finds the first element matching the specified attributes.
```

Comment:
How about making it a bit more explicit that *all* attributes should match?

```suggestion
  # Finds the first element matching all the specified attributes.
```